### PR TITLE
[TEST-ONLY]

### PR DESCRIPTION
**Pull request by ReToCode**
Additional stuff to https://github.com/openshift-knative/serverless-operator/pull/2126 for testing.

/hold

### CODE REVIEW

File: hack/lib/mesh_resources/peerauthentication.yaml [hunk 1]
```diff
@@ -6,93 +6,3 @@ metadata:
...
-  namespace: knative-eventing
-spec:
-  mtls:
-    mode: STRICT
----
-apiVersion: security.istio.io/v1beta1
-kind: PeerAuthentication
-metadata:
-  name: default
-  namespace: eventing-e2e0
```
Comment (mgencur) [inline hack/lib/mesh_resources/peerauthentication.yaml:32]: Just a note. `eventing-e2e0` - `eventing-e2e4` are namespaces used by upgrade tests. These tests do not run by default in CI with Service Mesh. We might need to define some rules for these namespaces as well.

Comment (ReToCode) [reply hack/lib/mesh_resources/peerauthentication.yaml:32]: The working theory is, that adding it to `istio-system` will apply it to the whole mesh automatically. At least upstream istio does that and documents it. I'll ask the SM team if we can also use that.

**Comment by ReToCode** [comment]
Still 503s, but not really consistent Oo:

```
    stream.go:305: E 08:22:51.418 activator-67cf5b64d8-kp2pd [activator] [serving-tests/revision-timeout-does-not-excee-wfxtxfne-00001] Failed to probe clusterIP 172.30.242.135:80 err=unexpected status code: want [200], got 503
```
/test 4.12-upstream-e2e-mesh-aws-ocp-412

**Comment by ReToCode** [comment]
/test 4.12-upstream-e2e-mesh-aws-ocp-412

```
tests/b-y-o-certificate-qwjpahix-00001 (depth: 1)
    stream.go:305: W 09:46:23.913 activator-67cf5b64d8-bdhsd [activator] [serving-tests/b-y-o-certificate-qwjpahix-00001] Failed probing pods err=unexpected status code: want [200], got 503
```

**Comment by ReToCode** [comment]
Local run

```
Running go test with args: -race -count=1 -tags=e2e -failfast -timeout=60m -parallel=1 ./test/servinge2e/ ./test/servinge2e/servicemesh/ --kubeconfigs /Users/rlehmann/Documents/kubeconfig-noingress,/Users/rlehmann/code/openshift-knative/serverless-operator/user3.kubeconfig,/Users/rlehmann/code/openshift-knative/serverless-operator/user1.kubeconfig,/Users/rlehmann/code/openshift-knative/serverless-operator/user2.kubeconfig --imagetemplate {{- with .Name }}
{{- if eq . "httpproxy" }}quay.io/openshift-knative/serving/{{.}}:v1.9
{{- else if eq . "recordevents" }}quay.io/openshift-knative/eventing/{{.}}:v1.9
{{- else if eq . "wathola-forwarder" }}quay.io/openshift-knative/eventing/{{.}}:v1.9
{{- else if eq . "kafka" }}strimzi/kafka:0.16.2-kafka-2.4.0
{{- else }}quay.io/openshift-knative/{{.}}:multiarch{{end -}}
{{end -}}
PASS test/servinge2e.TestKnConsoleCLIDownload (0.13s)
PASS test/servinge2e
PASS test/servinge2e/servicemesh.TestMultiTenancyWithServiceMesh/cross-tenant-via-ingress-via-activator (30.43s)
PASS test/servinge2e/servicemesh.TestMultiTenancyWithServiceMesh/cross-tenant-via-activator (20.47s)
PASS test/servinge2e/servicemesh.TestMultiTenancyWithServiceMesh/cross-tenant-via-ingress-no-activator (20.30s)
PASS test/servinge2e/servicemesh.TestMultiTenancyWithServiceMesh/cross-tenant-directly (20.29s)
PASS test/servinge2e/servicemesh.TestMultiTenancyWithServiceMesh/same-tenant-via-ingress-via-activator (20.23s)
PASS test/servinge2e/servicemesh.TestMultiTenancyWithServiceMesh/same-tenant-via-ingress-no-activator (20.26s)
PASS test/servinge2e/servicemesh.TestMultiTenancyWithServiceMesh/same-tenant-via-activator (20.30s)
PASS test/servinge2e/servicemesh.TestMultiTenancyWithServiceMesh/same-tenant-directly (20.25s)
PASS test/servinge2e/servicemesh.TestMultiTenancyWithServiceMesh (0.00s)
PASS test/servinge2e/servicemesh

=== Skipped
=== SKIP: test/servinge2e TestTraceStartedAtActivator (0.18s)
time="2023-07-07T11:23:15+02:00" level=info msg="Loading kube client config from path \"/Users/rlehmann/Documents/kubeconfig-noingress\""
time="2023-07-07T11:23:15+02:00" level=info msg="Loading kube client config from path \"/Users/rlehmann/code/openshift-knative/serverless-operator/user3.kubeconfig\""
time="2023-07-07T11:23:15+02:00" level=info msg="Loading kube client config from path \"/Users/rlehmann/code/openshift-knative/serverless-operator/user1.kubeconfig\""
time="2023-07-07T11:23:15+02:00" level=info msg="Loading kube client config from path \"/Users/rlehmann/code/openshift-knative/serverless-operator/user2.kubeconfig\""
    tracing_test.go:46: ServiceMesh installed, skipping tracing test.

=== SKIP: test/servinge2e TestTraceStartedAtQueueProxy (0.00s)
    tracing_test.go:38: Activator is always on the path. See SRVKS-784

DONE 12 tests, 2 skipped in 177.791s
Finished run, return code is 0
XML report written to /var/folders/cr/k3mnt3d56_v06ywwstjgkywr0000gn/T/tmp.Ee4Z2Jm3a1/junit_jjkI4iuM.xml
Test log (JSONL) written to /var/folders/cr/k3mnt3d56_v06ywwstjgkywr0000gn/T/tmp.Ee4Z2Jm3a1/go_test_jjkI4iuM.jsonl
Test log (ANSI) written to /var/folders/cr/k3mnt3d56_v06ywwstjgkywr0000gn/T/tmp.Ee4Z2Jm3a1/go_test_jjkI4iuM-ansi.log
Test log (HTML) written to /var/folders/cr/k3mnt3d56_v06ywwstjgkywr0000gn/T/tmp.Ee4Z2Jm3a1/go_test_jjkI4iuM.html
SUCCESS 11:26:11.189 🌟 Tests have passed 🌟
```

```
EMPTY test/conformance/api/shared
PASS test/conformance/api/v1.TestServiceAccountValidation (0.12s)
PASS test/conformance/api/v1.TestConfigurationGetAndList (19.42s)
PASS test/conformance/api/v1.TestServiceCreateListAndDelete (20.41s)
PASS test/conformance/api/v1.TestRouteGetAndList (20.25s)
PASS test/conformance/api/v1.TestRevisionGetAndList (20.55s)
PASS test/e2e/domainmapping.TestBYOCertificate (52.65s)
PASS test/conformance/api/v1.TestServiceBYOName (55.06s)
PASS test/e2e/domainmapping
PASS test/conformance/api/v1.TestAnnotationPropagation (57.73s)
PASS test/conformance/api/v1.TestSecretVolume (65.61s)
PASS test/conformance/api/v1.TestCustomResourcesLimits (66.84s)
PASS test/conformance/api/v1.TestProjectedSecretVolume (70.58s)
PASS test/e2e/initcontainers.TestInitContainers (70.33s)
PASS test/conformance/runtime.TestShouldHaveStdinEOF (69.41s)
PASS test/e2e/pvc.TestPersistentVolumeClaims (71.26s)
PASS test/conformance/runtime.TestMustNotContainerConstraints/TestArbitraryPortName (0.40s)
PASS test/conformance/api/v1.TestConfigMapVolume (71.14s)
PASS test/conformance/runtime.TestMustNotContainerConstraints/TestMountPropagation (0.50s)
PASS test/conformance/runtime.TestMustNotContainerConstraints (0.01s)
PASS test/conformance/api/v1.TestServiceCreateWithEmptyDir (71.33s)
PASS test/conformance/api/v1.TestProjectedServiceAccountToken (71.38s)
PASS test/e2e/initcontainers
PASS test/e2e/pvc
PASS test/conformance/api/v1.TestServiceCreateWithMultipleContainers (75.33s)
PASS test/conformance/api/v1.TestProjectedConfigMapVolume (75.54s)
PASS test/conformance/api/v1.TestSingleConcurrency (85.13s)
PASS test/conformance/api/v1.TestUpdateConfigurationMetadata (29.41s)
PASS test/e2e.TestDestroyPodTimely (104.68s)
PASS test/conformance/api/v1.TestServiceGenerateName (61.28s)
PASS test/conformance/api/v1.TestRouteAndConfigGenerateName (68.26s)
PASS test/conformance/runtime.TestShouldHaveCgroupReadOnly (79.03s)
PASS test/conformance/runtime.TestShouldEnvVars (83.56s)
PASS test/conformance/runtime.TestMustRunAsUser (83.57s)
PASS test/conformance/runtime.TestShouldRunAsUserContainerDefault (83.65s)
PASS test/conformance/runtime.TestShouldHaveSysctlReadOnly (83.84s)
PASS test/conformance/runtime.TestMustEnvVars (83.85s)
PASS test/conformance/runtime.TestProtocols/http1 (83.97s)
PASS test/conformance/runtime.TestWorkingDirService (88.57s)
PASS test/conformance/runtime.TestProtocols/default (88.42s)
PASS test/conformance/runtime.TestProtocols (0.00s)
PASS test/conformance/runtime.TestMustHaveCgroupConfigured (88.88s)
PASS test/conformance/runtime.TestShouldHaveHeadersSet (89.19s)
PASS test/conformance/api/v1.TestContainerExitingMsg (122.11s)
PASS test/conformance/runtime.TestMustHaveFiles (93.73s)
PASS test/conformance/runtime.TestShouldNotContainerConstraints/TestPrestopHook (0.10s)
PASS test/conformance/runtime.TestShouldNotContainerConstraints/TestHostPort (0.07s)
PASS test/conformance/runtime.TestShouldNotContainerConstraints/TestInvalidUID (0.08s)
PASS test/conformance/runtime.TestShouldNotContainerConstraints/TestStdinOnce (0.15s)
PASS test/conformance/runtime.TestShouldHaveFiles (94.14s)
PASS test/conformance/runtime.TestCmdArgs (94.20s)
PASS test/conformance/runtime.TestShouldNotContainerConstraints/TestMultiplePorts (0.19s)
PASS test/conformance/runtime.TestShouldNotContainerConstraints/TestStdin (0.16s)
PASS test/conformance/runtime.TestShouldNotContainerConstraints/TestTTY (0.25s)
PASS test/conformance/runtime.TestMustHaveHeadersSet (94.41s)
PASS test/conformance/runtime.TestShouldNotContainerConstraints/TestPoststartHook (0.08s)
PASS test/conformance/runtime.TestShouldNotContainerConstraints (0.01s)
PASS test/conformance/api/v1.TestRouteCreation (126.84s)
PASS test/conformance/api/v1.TestRevisionTimeout/exceeds_timeout_seconds (99.41s)
PASS test/conformance/api/v1.TestRevisionTimeout/does_not_exceed_timeout_seconds (100.36s)
PASS test/conformance/runtime.TestProbeRuntime/httpGet (108.39s)
PASS test/conformance/api/v1.TestProjectedComplex (179.81s)
PASS test/conformance/api/v1.TestRevisionTimeout/writes_first_byte_before_timeout (113.63s)
PASS test/conformance/api/v1.TestRevisionTimeout (0.01s)
PASS test/conformance/api/v1.TestBlueGreenRoute (151.06s)
PASS test/conformance/api/v1.TestServiceCreateAndUpdate (221.74s)
PASS test/conformance/runtime.TestProbeRuntime/tcpSocket-period=1 (81.43s)
PASS test/conformance/runtime.TestProbeRuntime/exec (82.24s)
PASS test/conformance/runtime.TestProbeRuntime/httpGet-period=1 (87.34s)
PASS test/e2e.TestDestroyPodWithRequests (138.92s)
PASS test/conformance/api/v1.TestServiceWithTrafficSplit (244.61s)
PASS test/conformance/api/v1
PASS test/conformance/runtime.TestSecretsViaEnv/envFrom (91.47s)
PASS test/conformance/runtime.TestProbeRuntime/tcpSocket (92.39s)
PASS test/conformance/runtime.TestSecretsViaEnv/env (97.57s)
PASS test/conformance/runtime.TestSecretsViaEnv (0.01s)
PASS test/conformance/runtime.TestProbeRuntime/exec-period=1 (102.07s)
PASS test/conformance/runtime.TestProbeRuntime (0.01s)
PASS test/conformance/runtime.TestConfigsViaEnv/env (97.32s)
PASS test/conformance/runtime.TestConfigsViaEnv/envFrom (97.26s)
PASS test/conformance/runtime.TestConfigsViaEnv (0.02s)
PASS test/e2e.TestWebSocket (18.24s)
PASS test/conformance/api/v1beta1.TestDomainMapping (271.10s)
PASS test/conformance/api/v1beta1
PASS test/e2e.TestWebSocketViaActivator (15.51s)
PASS test/conformance/runtime.TestProbeRuntimeAfterStartup/periodSeconds=1 (100.31s)
PASS test/conformance/runtime.TestProbeRuntimeAfterStartup/periodSeconds=0 (120.44s)
PASS test/conformance/runtime.TestProbeRuntimeAfterStartup (0.00s)
PASS test/conformance/runtime
PASS test/conformance/api/v1alpha1.TestDomainMapping (281.09s)
PASS test/conformance/api/v1alpha1
PASS test/e2e.TestWebSocketBlueGreenRoute (23.39s)
PASS test/e2e.TestWebSocketWithTimeout/does_not_exceed_timeout_seconds (10.18s)
PASS test/e2e.TestWebSocketWithTimeout/exceeds_timeout_seconds (17.20s)
PASS test/e2e.TestWebSocketWithTimeout/does_not_exceed_response_start_timeout_seconds (10.23s)
PASS test/e2e.TestWebSocketWithTimeout/exceeds_response_start_timeout_seconds (17.92s)
PASS test/e2e.TestWebSocketWithTimeout/does_not_exceed_idle_timeout_seconds (9.25s)
PASS test/e2e.TestWebSocketWithTimeout/exceeds_idle_timeout_seconds (16.17s)
PASS test/e2e.TestWebSocketWithTimeout (80.97s)
PASS test/e2e.TestRoutesNotReady (2.20s)
PASS test/e2e.TestPodScheduleError (4.12s)
PASS test/e2e.TestSubrouteVisibilityPrivateToPublic (29.33s)
PASS test/e2e.TestSubrouteVisibilityPublicToPrivate (32.36s)
PASS test/e2e.TestResourceQuotaError (39.11s)
PASS test/e2e.TestConflictingRouteService (35.30s)
PASS test/e2e.TestServiceValidationWithInvalidPodSpec (0.30s)
PASS test/e2e.TestHelloWorld (60.37s)
PASS test/e2e.TestEmptyDir (60.43s)
PASS test/e2e.TestInitScalePositive (61.21s)
PASS test/e2e.TestMultiContainer (61.28s)
PASS test/e2e.TestQueueSideCarResourceLimit (65.60s)
PASS test/e2e.TestEgressTraffic (70.37s)
PASS test/e2e.TestRollbackBYOName (72.54s)
FAIL test/e2e.TestDestroyPodInflight (94.56s)
PASS test/e2e.TestRequestLogs (70.61s)
PASS test/e2e.TestRouteVisibilityChanges/Route_visibility_changes_from_public_to_private_with_both_tagged_and_non-tagged_traffic (44.42s)
PASS test/e2e.TestRouteVisibilityChanges/Route_visibility_changes_from_public_to_private_with_single_traffic (54.49s)
PASS test/e2e.TestRouteVisibilityChanges/Route_visibility_changes_from_public_to_private_with_tag_only (50.44s)
PASS test/e2e.TestRouteVisibilityChanges (0.01s)
PASS test/e2e.TestGradualRollout (115.54s)
PASS test/e2e.TestImagePullError (122.11s)
PASS test/e2e.TestReadinessAlternatePort (84.31s)
PASS test/e2e.TestMinScale (112.54s)
PASS test/e2e.TestMultipleNamespace (123.91s)
PASS test/e2e.TestRevisionTimeout/writes_response_before_idle_timeout (90.45s)
PASS test/e2e.TestRevisionTimeout/exceeds_idle_timeout (95.68s)
PASS test/e2e.TestRevisionTimeout/writes_response_before_response_start_timeout (96.38s)
PASS test/e2e.TestRevisionTimeout/exceeds_response_start_timeout (107.81s)
PASS test/e2e.TestRevisionTimeout/exceeds_timeout_seconds (97.34s)
PASS test/e2e.TestRevisionTimeout/does_not_exceed_timeout_seconds (82.10s)
PASS test/e2e.TestRevisionTimeout (0.01s)
PASS test/e2e.TestServiceToServiceCall/short (97.29s)
PASS test/e2e.TestServiceToServiceCall/shortest (105.45s)
PASS test/e2e.TestServiceToServiceCall/fqdn (110.42s)
PASS test/e2e.TestServiceToServiceCall (20.14s)
PASS test/e2e.TestCallToPublicService/local_address (102.24s)
PASS test/e2e.TestSubrouteLocalSTS/shortest (104.19s)
PASS test/e2e.TestCallToPublicService/external_address (80.59s)
PASS test/e2e.TestCallToPublicService (24.17s)
PASS test/e2e.TestSubrouteLocalSTS/short (112.27s)
PASS test/e2e.TestSubrouteLocalSTS/fqdn (117.00s)
PASS test/e2e.TestSubrouteLocalSTS (22.15s)
...
DONE 143 tests, 2 skipped, 1 failures in 795.182s
```

Need to dig a bit more to check
```
FAIL test/e2e.TestDestroyPodInflight (94.56s)
```

File: hack/lib/mesh_resources/smcp.yaml [hunk 2]
```diff
@@ -4,6 +4,10 @@ metadata:
   name: basic
   namespace: istio-system
 spec:
+  techPreview:
+    meshConfig:
+      defaultConfig:
+        terminationDrainDuration: 35s
```
Comment (mgencur) [inline hack/lib/mesh_resources/smcp.yaml:10]: Just FYI, we also deploy this webhook for mesh tests which sets the timeout to 20 seconds: https://github.com/openshift-knative/serverless-operator/blob/main/serving/metadata-webhook/pkg/defaults/ksvc_defaults.go#L19
It probably overrides what you define here.

**Comment by ReToCode** [comment]
This is now using an [updated version](https://github.com/openshift-knative/net-istio/compare/release-v1.9...ReToCode:net-istio:peer-auth?expand=1) of `net-istio`

File: hack/lib/mesh_resources/smcp.yaml (hunk 2 above)
Comment (ReToCode) [reply hack/lib/mesh_resources/smcp.yaml:10]: Yeah I've seen that, but it does not seem to work. Envoy still terminates after seconds. That flag seems to work. We'll need to dig into that as well.

Comment (ReToCode) [reply hack/lib/mesh_resources/smcp.yaml:10]: From https://gist.github.com/ReToCode/e0b2a8d0d91809bf489caa6e26d287ca#file-envoy-proxy-terminates-too-early-txt
```
2023-07-13T09:10:28.706951023Z istio-proxy 2023-07-13T09:10:28.706875Z    warn    Envoy proxy is NOT ready: server is not live, current state is: DRAINING
2023-07-13T09:10:30.707229025Z istio-proxy 2023-07-13T09:10:30.707137Z    warn    Envoy proxy is NOT ready: server is not live, current state is: DRAINING
2023-07-13T09:10:32.706613579Z istio-proxy 2023-07-13T09:10:32.706571Z    warn    Envoy proxy is NOT ready: server is not live, current state is: DRAINING
2023-07-13T09:10:32.768848101Z istio-proxy 2023-07-13T09:10:32.768775Z    warn    Aborted proxy instance
2023-07-13T09:10:32.768848101Z istio-proxy 2023-07-13T09:10:32.768811Z    warn    Aborting proxy
```

Comment (mgencur) [reply hack/lib/mesh_resources/smcp.yaml:10]: Ah. That was the DestroyPodInflight failure. Thanks for the info.

Comment (ReToCode) [reply hack/lib/mesh_resources/smcp.yaml:10]: 👍 more details here: https://issues.redhat.com/browse/SRVKS-1099

**Comment by ReToCode** [comment]
Hm [first](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/2138/pull-ci-openshift-knative-serverless-operator-main-4.13-upstream-e2e-mesh-aws-ocp-413/1680895468877385728) try was better, the [second](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/2138/pull-ci-openshift-knative-serverless-operator-main-4.13-upstream-e2e-mesh-aws-ocp-413/1680915179056926720) has new flakes, that are not there locally:

```
PASS test/e2e.TestRevisionTimeout/exceeds_timeout_seconds (15.21s)
PASS test/e2e.TestRevisionTimeout/exceeds_response_start_timeout (12.23s)
PASS test/e2e.TestRevisionTimeout/exceeds_idle_timeout (12.16s)
PASS test/e2e.TestRevisionTimeout/writes_response_before_idle_timeout (7.14s)
PASS test/e2e.TestRevisionTimeout/does_not_exceed_timeout_seconds (7.21s)
PASS test/e2e.TestRevisionTimeout/writes_response_before_response_start_timeout (9.17s)
PASS test/e2e.TestRevisionTimeout (0.06s)
PASS test/e2e
```

Moore digging...

**Comment by ReToCode** [comment]
[Eventing test flakes.](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/2138/pull-ci-openshift-knative-serverless-operator-main-4.13-upstream-e2e-mesh-aws-ocp-413/1680941793752911872)

/test 4.13-upstream-e2e-mesh-aws-ocp-413

**Comment by ReToCode** [comment]
Closing this approach in favour of the one with `mtls: true` config: 
https://github.com/openshift-knative/serverless-operator/pull/2171

**Pull request closed without merging**